### PR TITLE
First release

### PR DESCRIPTION
**Pull request by lazcanoluca**
This pull request merges the development branch with the progress up to date. The following features have been implemented and tested:

- [x] A Merkle Tree can be built out of an array.
- [x] A Merkle Tree can generate a proof that it contains an element.
- [x] A Merkle Tree can validate that the proof for a given hash is correct.
- [x] A Merkle Tree can verify that a given hash is contained in it.
- [x] A Merkle Tree can be dynamic, this means that elements can be added once it is built.

### CODE REVIEW

File: src/merkle_tree.rs [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+    use super::*;
+
+    #[test]
+    fn test_hash_should_return_sha256_digest() {
+        let input = "In a hole in the ground there lived a hobbit.";
+        let hash = MerkleTree::hash(input.as_bytes());
+
+        assert_eq!(
+            hash.to_vec(),
+            hex::decode("38a76005681abd4a4f50a364d472016436f17e79778577ee5825580f06997202")
```
Comment (belfortep) [inline src/merkle_tree.rs:194]: Here in the tests could be clearer if the different hashes are in variables

File: src/merkle_tree.rs [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+        Some(
+            parent_level
+                .chunks_exact(2)
+                .map(Self::merkle_parent)
+                .collect(),
+        )
+    }
+
+    /// Computes the Merkle root hash for the provided leaf hashes.
+    pub fn root(&self) -> Hash {
```
Comment (belfortep) [inline src/merkle_tree.rs]: could return an `Option<Hash>` to skip the unwraps

**Review by belfortep — APPROVED** [review]
The code looks great overall, maybe the places where there are unwraps could return an `Option<>` instead.

File: src/merkle_tree.rs (hunk 1 above)
Comment (lazcanoluca) [reply src/merkle_tree.rs:194]: Got it! Will do.

File: src/merkle_tree.rs (hunk 2 above)
Comment (lazcanoluca) [reply src/merkle_tree.rs]: Good point. Will do.

**Review by belfortep — APPROVED** [review]
Lgtm!

**Pull request merged**